src/main.rs

```diff
@@ -106,12 +110,32 @@ fn main() -> Result<(), Box<dyn Error>> {
             .arg("log")
             .arg("--numstat")
             .output()?;
+
         String::from_utf8(process.stdout)?
     } else if matches.opt_present("input") {
         let path = matches.opt_str("input").unwrap();
+
         std::fs::read_to_string(path)?
+    } else if matches.opt_present("stdin") {
+        let mut input = String::new();
+        loop {
+            let mut buffer = String::new();
+            match std::io::stdin().read_line(&mut buffer) {
+                Ok(0) => {
+                    break;
+                }
+                Ok(_) => {
+                    input = input + &buffer;
+                }
+                Err(error) => {
+                    return Err(error.into());
```

Comment:
This is exactly what the `?` operator would do. So you could write this match also like this:

```rust
match std::io::stdin().read_line(&mut buffer)? {
    0 => break,
    _ => input = input + &buffer,
}
```
which you then could also write like this:
```rust
if std::io::stdin().read_line(&mut buffer)? == 0 {
    break;
}
input = input + &buffer;
```
Note that `input + &buffer` may behave a little different, then in other languages, since afaik it actually changes the left hand side of the addition. So rewriting it as `input += &buffer` or even `input.push_str(&buffer)` could be considered. I _think_ the last one is the most "rusty" one. 😄

Reply:
:wrench: :heavy_check_mark: